promotion/datastore.go

```diff
@@ -1807,6 +1817,71 @@ func (pg *Postgres) UpdateDrainJobAsRetriable(ctx context.Context, walletID uuid
 	return nil
 }
 
+// RunNextGeminiCheckStatus periodically check the status of gemini claim drain transactions
+func (pg *Postgres) RunNextGeminiCheckStatus(ctx context.Context, worker GeminiTxnStatusWorker) (bool, error) {
+	tx, err := pg.RawDB().Beginx()
+	if err != nil {
+		return false, fmt.Errorf("gemini check status job: failed to begin transaction: %w", err)
+	}
+	defer pg.RollbackTx(tx)
+
+	var drainJob DrainJob
+	err = tx.Get(&drainJob, `
+									select * from claim_drain 
+									where status = $1 and transaction_id is not null
```

Comment:
Is this for the first check or for every status check ? If we alter this and add something like updated_at < NOW() - INTERVAL '10 MINUTES' that would mean we check the status of a tx no less than every 10 mins unless we have a concept of created_at to wait before the first check. We update the update_at after every status check so that we cycle through the pending jobs and dont run the risk of select the same pending job(s) every time. New jobs join the back of the queue.

Reply:
added an interval and can revisit if needed